### PR TITLE
DRAFT empty constant issue with luci-interpreter

### PR DESCRIPTION
**Pull request by seanshpark**
on-going draft to fix empty constant issue with luci-interpreter

ONE-DCO-1.0-Signed-off-by: SaeHie Park <saehie.park@gmail.com>

### CODE REVIEW

**Comment by seanshpark** [comment]
For #2968

File: compiler/luci-interpreter/src/core/Tensor.cpp [hunk 1]
```diff
@@ -36,6 +36,8 @@ void Tensor::readData(void *data_ptr, size_t data_size) const
 {
   const size_t element_size = getDataTypeSize(element_type());
   const int32_t num_elements = shape().num_elements();
+  if (num_elements <= 0)
```
Comment (s-barannikov) [inline compiler/luci-interpreter/src/core/Tensor.cpp:39]: I would instead assert that `data_ptr != nullptr && data_size != 0` after checking `data_size != num_elements * element_size` below. IMHO it is incorrect to call this function for empty tensor.

File: compiler/luci-interpreter/src/core/Tensor.cpp [hunk 2]
```diff
@@ -47,6 +49,8 @@ void Tensor::writeData(const void *data_ptr, size_t data_size)
...
```
Comment (s-barannikov) [inline compiler/luci-interpreter/src/core/Tensor.cpp:52]: Same

File: compiler/luci-interpreter/src/core/Tensor.cpp [hunk 3]
```diff
@@ -59,6 +63,8 @@ void Tensor::resize(const Shape &new_shape)
   _shape = new_shape;
   const size_t element_size = getDataTypeSize(_element_type);
   const int32_t num_elements = _shape.num_elements();
+  if (num_elements <= 0)
+    return;
```
Comment (s-barannikov) [inline compiler/luci-interpreter/src/core/Tensor.cpp:67]: Could you please add a comment near the `_data` field that it will be nullptr for empty tensors?

**Comment by seanshpark** [comment]
@s-barannikov , thanks for the review and I've update the change. please take a look again :)

**Comment by seanshpark** [comment]
all landed

**Pull request closed without merging**